src/google_form_to_engagement_db/google_form_to_engagement_db.py

```diff
@@ -170,6 +183,54 @@ def _form_answer_to_engagement_db_message(form_answer, form_id, form_response, p
     )
 
 
+def _all_equal(iterable):
+    """Checks if all elements in an iterable are identical"""
+    iterator = iter(iterable)
+    try:
+        first_item = next(iterator)
+    except StopIteration:
+        return True
+    return all(first_item == x for x in iterator)
+
+
+def _merge_engagement_db_messages(messages, messages_origin_details, answers_delimeter):
```

Comment:
Do the messages + messages origin details need to be in the same order? I wonder if it would be better to take a list of (message, origin_details) so you can guarantee the texts and origin ids are in the same order when you merge them

Reply:
Yes. I have stored the variables in one variable to avoid inconsistencies. Thanks.